inference_cli/lib/cloud_adapter.py

```diff
@@ -0,0 +1,202 @@
+import sky
+import random
+import string
+import tempfile
+
+YAML_DEFS = {
+"gcp_cpu": """
+name: roboflow-gcp-inference-cpu
+num_nodes: 1
+resources:
+  cloud: gcp
+  instance_type: n2-standard-2
+  use_spot: false
+  disk_size: 100
+  disk_tier: medium
+  ports: 9001
+  image_id: projects/ubuntu-os-cloud/global/images/ubuntu-2204-jammy-v20240112
+setup: |
+  sudo apt-get update
+  sudo apt-get install -y uidmap
+  curl -fsSL https://get.docker.com/rootless -o get-docker.sh
+  sh get-docker.sh
+  sudo loginctl enable-linger gcpuser
+  systemctl --user start docker.service
+  docker pull roboflow/roboflow-inference-server-cpu
+  export PATH=/home/gcpuser/bin:$PATH
+  echo
+run: |
+  export PATH=/home/gcpuser/bin:$PATH; docker run -d -p 9001:9001 PLACEHOLDER roboflow/roboflow-inference-server-cpu
+""",
+
+"gcp_gpu": """
+name: roboflow-gcp-inference-gpu
+num_nodes: 1
+resources:
+  cloud: gcp
+  accelerators: T4:1
+  instance_type: n1-standard-4
+  use_spot: false
+  disk_size: 100
+  disk_tier: medium
+  ports: 9001
+  image_id: projects/ml-images/global/images/c0-deeplearning-common-gpu-v20231209-debian-11
+setup: |
+  docker pull roboflow/roboflow-inference-server-gpu
+run: |
+  docker run -d --gpus all -p 9001:9001 PLACEHOLDER roboflow/roboflow-inference-server-gpu
```

Comment:
Is this PLACEHOLDER left here intentionally?

Reply:
It gets substituted in code